st2actions/st2actions/runners/remote_script_runner.py

```diff
@@ -78,3 +84,124 @@ def _get_remote_action(self, action_parameters):
                                         sudo=self._sudo,
                                         timeout=self._timeout,
                                         cwd=self._cwd)
+
+
+class ParamikoRemoteScriptRunner(BaseParallelSSHRunner):
+    def run(self, action_parameters):
+        remote_action = self._get_remote_action(action_parameters)
+
+        LOG.debug('Executing remote action.', extra={'_action_params': remote_action})
+        result = self._run(remote_action)
+        LOG.debug('Executed remote action.', extra={'_result': result})
+        status = self._get_result_status(result, cfg.CONF.ssh_runner.allow_partial_failure)
+
+        return (status, result, None)
+
+    def _run(self, remote_action):
+        try:
+            copy_results = self._copy_artifacts(remote_action)
+        except:
+            # If for whatever reason there is a top level exception,
+            # we just bail here.
+            error = 'Failed copying content to remote boxes.'
+            LOG.exception(error)
+            _, ex, tb = sys.exc_info()
+            copy_results = self._generate_error_results(' '.join([error, str(ex)]), tb)
+            return copy_results
+
+        try:
+            exec_results = self._run_script_on_remote_host(remote_action)
+            try:
+                remote_dir = remote_action.get_remote_base_dir()
+                LOG.debug('Deleting remote execution dir.', extra={'_remote_dir': remote_dir})
```

Comment:
Just FYI - you don't need to prefix extra keys with underscore anymore (this is already done inside the decorators which wrap the logger methods).